packages/i18n/src/locales/en.i18n.json

```
@@ -3631,6 +3631,7 @@
   "Marketplace_Invalid_Apps_Engine_Version": "The installed Apps Engine version is not compatible with the marketplace. Please update the Apps Engine to the latest version.",
   "Marketplace_Internal_Error": "An internal error occurred communicating with Marketplace. Please try again later.",
   "Marketplace_Failed_To_Fetch_Apps": "Failed to fetch apps from the marketplace. Please try again later.",
+  "Marketplace_Core_Schema_Missmatch": "Invalid apps were found in the marketplace. If the problem persists please contact the support.",
```

```suggestion
  "Marketplace_Core_Schema_Missmatch": "Invalid apps were found in the marketplace. If the problem persists please contact support.",
```

A customer may not know what an "invalid app" is. Feel this is really an internal bug. Can we modify the message to something ore user friendly? Like
"Error fetching apps from marketplace" or so.

---

I guess we could be more generic, but I this is not handled in the frontend yet, is just the backend saying what has happened.
Not necessarily customer faced

---

Yeah but you're translating and sending the error to the customer. When they face a similar situation, this is what they'll see on the network and may be weird for them.

My suggestion is to change it to be generic to the customer and detailed on error logs (as non-tech people may access marketplace)

---

I guess, not sure many non-tech people would access the network tab though. What concerns me is that the users are going to see a generic error message that is going to give then 0 guidance about what to do.

One of the main points of the initiative is to make it so the users don't get blasted with generic, Apps_Error_*-like messages whenver something goes wrong.

I think that in the near future, we could even do something like, validate the schema of all the apps, and only render the ones that don't have an error, if one has the error, we could call marketplace-api to mark it as invalid, notify the developer and set the status as invalid or something, so we can avoid fetching it next time.

---

Yeah but it makes sense to give them 0 guidance cause this issue is not something they can solve...

---

But I can't see why just popping a generic error message would be better. Feels like this way the user that already knows how to access the network tab an look for errors there should contact the support so we can go and fix it.
Also, this error will probably be always catched by us, not by end-users, since it will happen usually when we change the marketplace API and test against the core